cool-core/src/test/java/com/nus/cool/functionality/CohortSelectionTest.java

```diff
@@ -0,0 +1,75 @@
+package com.nus.cool.functionality;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.nus.cool.core.cohort.ExtendedCohortQuery;
+import com.nus.cool.core.io.readstore.CubeRS;
+import com.nus.cool.model.CoolModel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.AfterTest;
+import org.testng.annotations.BeforeTest;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Arrays;
+import java.util.List;
+
+public class CohortSelectionTest extends CsvLoaderTest {
+    static final Logger logger = LoggerFactory.getLogger(CohortSelectionTest.class);
+
+    @BeforeTest
+    public void setUp() {
+        logger.info("Start UnitTest " + CohortSelectionTest.class.getSimpleName());
+    }
+
+    @AfterTest
+    public void tearDown() {
+        logger.info(String.format("Tear Down UnitTest %s\n", CohortSelectionTest.class.getSimpleName()));
+    }
+
+    @Test(dataProvider = "CohortSelectionTestDP", dependsOnMethods="CsvLoaderUnitTest")
+    public void CohortSelectionUnitTest(String datasetPath, String queryPath) throws IOException {
+
+        ObjectMapper mapper = new ObjectMapper();
+        ExtendedCohortQuery query = mapper.readValue(new File(queryPath), ExtendedCohortQuery.class);
+
+        String inputSource = query.getDataSource();
+        CoolModel coolModel = new CoolModel(datasetPath);
+        coolModel.reload(inputSource);
+
+        CubeRS cube = coolModel.getCube(query.getDataSource());
+
+        List<Integer> cohortResults = coolModel.cohortEngine.selectCohortUsers(cube,null, query);
+        List<Integer> selectionGlobalIDs = Arrays.asList(0, 2, 3, 4, 5, 7, 9, 11, 12);
+        assert cohortResults.equals(selectionGlobalIDs);
+        List<String> userIDs = coolModel.cohortEngine.listCohortUsers(cube, cohortResults);
+        List<String> selectionActualIDs = Arrays.asList("P-0", "P-2", "P-3", "P-4", "P-5", "P-7", "P-9", "P-11", "P-12");
```

Comment:
Pass in as a parameter, decoupling data and logic

Reply:
update